crates/hotshot/src/tasks/mod.rs

```diff
@@ -67,12 +68,16 @@ pub async fn add_network_message_task<TYPES: NodeType, NET: CommunicationChannel
         let network = net.clone();
         let closure = async move {
             loop {
-                let msgs = Messages(
-                    network
-                        .recv_msgs(TransmitType::Broadcast)
-                        .await
-                        .expect("Failed to receive broadcast messages"),
-                );
+                let msgs = match network.recv_msgs(TransmitType::Broadcast).await {
+                    Ok(msgs) => Messages(msgs),
+                    Err(err) => {
```

Comment:
Were we getting this error? Seems like this shouldn't happen on the receive side

Reply:
We were, but I believe it was from the nodes that were suppoed to be killed. This is actually fixed with the other changes I made, but I still don't think we should be panicking - WDYT?

Reply:
I think it's a good change, just would be quite surprised if we hit this outside of shutdown